openmc/deplete/chain.py

```diff
@@ -0,0 +1,786 @@
+                    type_ = ','.join(mode.modes)
+                    if mode.daughter in decay_data:
+                        target = mode.daughter
+                    else:
+                        print('missing {} {} {}'.format(
+                            parent, ','.join(mode.modes), mode.daughter))
+                        target = replace_missing(mode.daughter, decay_data)
+
+                    # Write branching ratio, taking care to ensure sum is unity
+                    br = mode.branching_ratio.nominal_value
+                    sum_br += br
+                    if i == len(data.modes) - 1 and sum_br != 1.0:
+                        br = 1.0 - sum(m.branching_ratio.nominal_value
+                                       for m in data.modes[:-1])
+
+                    # Append decay mode
+                    nuclide.decay_modes.append(DecayTuple(type_, target, br))
+
+            if parent in reactions:
+                reactions_available = set(reactions[parent].keys())
+                for name, mts, changes in _REACTIONS:
+                    if mts & reactions_available:
+                        delta_A, delta_Z = changes
+                        A = data.nuclide['mass_number'] + delta_A
+                        Z = data.nuclide['atomic_number'] + delta_Z
+                        daughter = '{}{}'.format(openmc.data.ATOMIC_SYMBOL[Z], A)
+
+                        if name not in chain.reactions:
+                            chain.reactions.append(name)
+
+                        if daughter not in decay_data:
+                            missing_rx_product.append((parent, name, daughter))
+
+                        # Store Q value
+                        for mt in sorted(mts):
+                            if mt in reactions[parent]:
+                                q_value = reactions[parent][mt]
+                                break
+                        else:
+                            q_value = 0.0
+
+                        nuclide.reactions.append(ReactionTuple(
+                            name, daughter, q_value, 1.0))
+
+                if any(mt in reactions_available for mt in [18, 19, 20, 21, 38]):
+                    if parent in fpy_data:
+                        q_value = reactions[parent][18]
+                        nuclide.reactions.append(
+                            ReactionTuple('fission', 0, q_value, 1.0))
+
+                        if 'fission' not in chain.reactions:
+                            chain.reactions.append('fission')
+                    else:
+                        missing_fpy.append(parent)
+
+            if parent in fpy_data:
+                fpy = fpy_data[parent]
+
+                if fpy.energies is not None:
+                    yield_energies = fpy.energies
+                else:
+                    yield_energies = [0.0]
+
+                yield_data = {}
+                for E, table in zip(yield_energies, fpy.independent):
```

Comment:
Optional: I would use a different identifier than `table`.  If it is the table of fission products (which I believe it to be by my read below), I would use `fission_products` or `fission_table`.